src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncMethodBuilderMemberCollection.cs

```diff
@@ -137,12 +137,19 @@ internal static bool TryCreate(SyntheticBoundNodeFactory F, MethodSymbol method,
             {
                 var returnType = (NamedTypeSymbol)method.ReturnType;
                 NamedTypeSymbol builderType;
-                MethodSymbol createBuilderMethod;
-                PropertySymbol taskProperty;
-                bool customBuilder = returnType.IsCustomTaskType(out builderType, out createBuilderMethod);
+                MethodSymbol createBuilderMethod = null;
+                PropertySymbol taskProperty = null;
+
+                object builderArgument;
+                bool customBuilder = returnType.IsCustomTaskType(out builderArgument);
                 if (customBuilder)
                 {
-                    taskProperty = GetCustomTaskProperty(F, builderType);
+                    builderType = ValidateBuilderType(F, builderArgument, returnType.DeclaredAccessibility, desiredArity:0);
+                    if (builderType != null)
```

Comment:
Fixed in https://github.com/dotnet/roslyn/pull/13405/commits/0e30e9088fbb4b80dd244a4c49944c1bbfefa3f0#diff-995d1bf33206eef8c980dbd08b8601b0R148
https://github.com/dotnet/roslyn/pull/13405/commits/0e30e9088fbb4b80dd244a4c49944c1bbfefa3f0#diff-995d1bf33206eef8c980dbd08b8601b0R216